params/config.go

```diff
@@ -544,11 +566,21 @@ func (c *ChainConfig) IsByzantium(num uint64) bool {
 	return isForked(c.ByzantiumBlock, num)
 }
 
+// IsByzantium returns whether num is either equal to the Byzantium fork block or greater.
+func (c *ChainConfig) IsByzantiumBigInt(num *big.Int) bool {
+	return configNumEqual(c.ByzantiumBlock, num)
+}
+
 // IsConstantinople returns whether num is either equal to the Constantinople fork block or greater.
 func (c *ChainConfig) IsConstantinople(num uint64) bool {
 	return isForked(c.ConstantinopleBlock, num)
 }
 
+// IsRamanujan returns whether num is either equal to the IsRamanujan fork block or greater.
+func (c *ChainConfig) IsRamanujanBigInt(num *big.Int) bool {
+	return configNumEqual(c.RamanujanBlock, num)
```

Comment:
This should be using `isForked` function instead of `configNumEqual` ?

Reply:
I will double check with what BSC did here

Reply:
I use `configNumEqual` here since the definitions are:
```
func isForked(s *big.Int, head uint64) bool
func configNumEqual(x, y *big.Int) bool
```

There is a compilation error when using `isForked`:
```
params/config.go:581:19: cannot use num (type *big.Int) as type uint64 in argument to isForked
```

Reply:
Perhaps in BSC, this hardfork is different, and instead of changing the rules after certain block, it has to execute something on that block, then I would understand the semantics. Can you confirm what Ramanujan hardfork is doing

Reply:
Here is what BSC is doing - https://github.com/binance-chain/bsc/blob/master/params/config.go#L732 and here is what is implemented in Erigon - https://github.com/covalenthq/erigon/blob/devel/params/config.go#L710. What I've added is basically a workaround to deal with the 2 forks using `big.Int` vs `uint64`.

This is all so I can do the checks in these locations: https://github.com/ledgerwatch/erigon/blob/8791d432c846c118e64239ba26e9e8d6416ef92c/consensus/parlia/ramanujanfork.go#L18 https://github.com/ledgerwatch/erigon/blob/8791d432c846c118e64239ba26e9e8d6416ef92c/consensus/parlia/ramanujanfork.go#L31 https://github.com/ledgerwatch/erigon/blob/8791d432c846c118e64239ba26e9e8d6416ef92c/consensus/parlia/ramanujanfork.go#L38